### ISSUE TITLE
Problem where interactable NPC won't show up in location despite having that dialog level

### ISSUE THREAD

**Issue by peachdoxie** [Korzer420/LoreRandomizer#2]
Twice now I've played randos where NPCs won't show up where they should, and both times they've had my progression and I'm unable to continue without using debug or benchwarp and breaking logic. The first was Hornet in Beast's Den and the second one is Quirrel in Deepnest. I suspect it might be because I'm missing certain movement (wings for Hornet, and claw for Quirrel) that would trigger the characters showing up in the right locations, but I'm not sure. I no longer have the logs for the Hornet game but have saved the ones for the Quirrel game if desired.

**Comment by Korzer420** [comment]
Hey sorry to hear that.

Could you provide your Modlog, the settings.txt file from that rando as well as your userX.modded.json file? 

All of these can be found in the AppData/LocalLow/Team Cherry/Hollow Knight directory (if you are on windows at least, not sure how the path is on other OS). 

The rando settings file is in Randomizer 4/Recent folder in that directory, in case that rando were this occured was your last one.

**Comment by peachdoxie** [comment]
[settings.txt](https://github.com/Korzer420/LoreRandomizer/files/12328531/settings.txt)
[ModLog 8-12.txt](https://github.com/Korzer420/LoreRandomizer/files/12328533/ModLog.8-12.txt)

Here are the settings file and the modlog. Github won't let me upload the user1.modded.json file, but I am not sure if it is useful anymore since I started a new rando in between submitting this issue and when I got your reply. What would you like me to do about that?

**Comment by peachdoxie** [comment]
Hi again. I had this same issue once more, where Quirrel in Deepnest didn't spawn and was my only in-logic check. I found the item in the spoiler log and used debug to get to it and continued playing that way. After two more Quirrel levels, I checked back in Deepnest and he was there, so I was able to get the item and continue without using debug. I don't know if this is useful but I thought I'd share what happened.

**Comment by Korzer420** [comment]
Hey, I couldn't reproduce the issue. I could add some kind of backup system that places a shiny there, although this might take a few days.

In case you do another attempt and encounter this issue please provide the location as well as the level of that traveller. If this doesn't occur with the higher level like you mentioned maybe there is a problem in the logic order. Either way you can modify the savedata to increase the traveller level to force the npc to appear (if that's the solution).

- Open userX.modded.json (X is the save file slot (from 1 to 4))
- Search for "Stages". This should bring you to a section that should look like this 
```
"LoreCore": {
      "Stages": {
        "Quirrel": 8,
        "Cloth": 4,
        "Tiso": 4,
        "Zote": 5,
        "Hornet": 5
      },
      "CanRead": true,
      "CanListen": true
    },
```
The number behind the npc name is the level, which you can then modify at will.
**Note that the game has to be closed for this**
- Save the file
- Restart the game

**Comment by peachdoxie** [comment]
Thank you for the solution if this occurs again. 

I went back through my tracker log and found the order in which I visited Quirrel and found his dialog, which I think is what you're asking for.

- Visited Quirrel in Crossroads
- Found Dialogue-Quirrel City at Emilitia (not at her lever)
- Visited Quirrel in Greenpath
- Found Dialogue-Quirrel Outside Archive at Geo Rock-Love Key
- Visited Quirrel in Queen Station
- Found Dialogue-Quirrel Peaks at Soul Totem-Deepnest Vessel
- Found Dialogue-Quirrel Blue Lake at Soul Totem-Oro Dive 1
- Visited Quirrel in City
- Found Dialogue-Quirrel Archive at Hornet Edge
- This is where my helper log and map showed Quirrel in Deepnest as my only in-logic check, but he did not appear so I used debug to clip through the wall he was supposed to give me (Howling Cliffs Breakable Near Cornifer). This gave me access to Broken Vessel, who provided the next Quirrel level (Dialogue-Quirrel Greenpath). However, everything was listed as "Accessible via sequence break" on the helper log reader.
- Visited Quirrel Peaks
- Visited Quirrel Outside Archive
- Found Dialogue-Quirrel Crossroads at Bench-Destroyed Tram
- Here, I checked back at Quirrel Deepnest and he was there, so everything that was out of logic via sequence break became in-logic.

If they're relevant, here's the rest of them:
- Found Dialogue-Quirrel Queen Station at Geo Rock-Crystal Peak Entrance Dupe 2
- Visited Quirrel in Mantis Village
- Found Dialogue-Quirrel Deepnest at Whispering Root-Queens Garden
- Visited Quirrel in Blue Lake
- Ended the rando.

**Comment by Korzer420** [comment]
@peachdoxie Hey, sorry for the late response. Was quite busy the last few months. I think I found the issue, the order in logic doesn't match the actual order in the code (at least for the Quirrel example, I'll review the other onces as well). I'll release an update in the next few days

**Comment by Korzer420** [comment]
Fixed in LoreRando 0.3 (+ LoreCore 0.6)

(Hornet had a different issue btw but this is fixed now)
